### PR TITLE
docs: Enterprise Use Case Tutorial - User Management, Access Controls, and Auditing with Feature Flags

### PR DESCRIPTION
**Pull request by nnennandukwe**
## About the changes

We're adding a new enterprise use case tutorial on user management, access controls, and auditing with feature flags.

All updates in this PR are changes made post-review of the [Google Doc](https://docs.google.com/document/d/1-snBfKxoCK-5ClchkIQVFK2YcIjWZjkbl5ENAB_S6zM/edit?tab=t.0).

Screenshot 2024-12-18 at 12 22 10 PM

Closes #

### CODE REVIEW

File: website/docs/feature-flag-tutorials/use-cases/user-management-access-controls.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+_Project roles_:
+
+1. Owner
+2. Member
+
+Assign users with root roles to configure entire systems. These users can perform any operation within the Unleash platform and change permissions for other users when they belong to a specific user group.
+
+Assign other users to project roles to have domain-specific control within a specific project maintained in Unleash.
+
+Project permissions are separated from root permissions to make it even more targeted regarding what permissions someone can and cannot have for each piece of Unleash. Assign developers with creation and modification rights and viewers who can observe but not change.
```
Comment (ferrantim) [inline website/docs/feature-flag-tutorials/use-cases/user-management-access-controls.md:96]: "for each piece of Unleash" is too vague. Can we be more specific?

File: website/docs/feature-flag-tutorials/use-cases/user-management-access-controls.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+For more fine-tuned access controls, create [custom root roles](/how-to/how-to-create-and-assign-custom-root-roles) and [custom project roles](/how-to/how-to-create-and-assign-custom-project-roles), where you can define the privileges and limitations beyond the predefined roles we have built into Unleash.
```
Comment (ferrantim) [inline website/docs/feature-flag-tutorials/use-cases/user-management-access-controls.md:98]: Can we give some examples of the types of access RBAC can be used for here?  I think there are something like 50+ things you can define. We could add a screenshot of that.

File: website/docs/feature-flag-tutorials/use-cases/user-management-access-controls.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+### Extend RBAC with a change management workflow
+
+While RBAC allows you to administer Unleash safely, you might need approval processes when changing feature flags or their configuration. When multiple teams are working on complex systems, [change requests](/reference/change-requests) provide a systematic approach to:
+
+-   **Comprehensive review**: Every proposed feature flag modification goes through a review, reducing the likelihood of unintended consequences.
+-   **Audit trail**: Every change is documented, timestamped, and attributed to specific team members, creating a permanent record of system modifications.
+-   **Four-eyes approval workflows**: Multi-stage approval processes ensure critical changes are thoroughly reviewed and approved by at least 2 other people before implementation.
```
Comment (ferrantim) [inline website/docs/feature-flag-tutorials/use-cases/user-management-access-controls.md:106]: Is it really that they are "approved by at least 2 **other** people before implementation."?  That would be 6 eyes. ;). I think, but am not sure, that you can have the requestor and a single approver.  Please confirm.

File: website/docs/feature-flag-tutorials/use-cases/user-management-access-controls.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+-   **Compliance requirements**: For regulated industries like finance and healthcare, Change Requests provide the detailed documentation necessary to meet strict compliance standards.
```
Comment (ferrantim) [inline website/docs/feature-flag-tutorials/use-cases/user-management-access-controls.md:107]: Let's link here to our new compliance docs section. https://docs.getunleash.io/using-unleash/compliance/compliance-overview

File: website/docs/feature-flag-tutorials/use-cases/user-management-access-controls.md [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+
+![Change requests are divided between two tabs: open and closed change request lists with relevant metadata listed per request.](/img/use-case-user-mgmt-change-requests.png)
```
Comment (ferrantim) [inline website/docs/feature-flag-tutorials/use-cases/user-management-access-controls.md:109]: This sentence is hard to parse. Can you revisit it please?

File: website/docs/feature-flag-tutorials/use-cases/user-management-access-controls.md [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+
+Imagine a large banking application where a development team wants to modify a feature flag controlling a new authentication method. Instead of a developer making an immediate change, the change request workflow might require:
```
Comment (ferrantim) [inline website/docs/feature-flag-tutorials/use-cases/user-management-access-controls.md:111]:
```suggestion
Imagine a developer working at our large banking platform wants to modify a feature flag controlling a new authentication method. Instead of a developer making an immediate change, the change request workflow might require:
```

File: website/docs/feature-flag-tutorials/use-cases/user-management-access-controls.md [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+
+1. Initial proposal submission
+2. Security team review
+3. Compliance officer approval
+4. Final sign-off from technical leadership
+
+![This GIF shows how to quickly make changes to your flag, request the change, approve it, and apply the changes.](/img/use-case-user-mgmt-cr.gif)
```
Comment (ferrantim) [inline website/docs/feature-flag-tutorials/use-cases/user-management-access-controls.md:118]:
```suggestion
![This GIF shows how to quickly make changes to your feature flag, request the change, approve it, and apply the changes.](/img/use-case-user-mgmt-cr.gif)
```

File: website/docs/feature-flag-tutorials/use-cases/user-management-access-controls.md [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+
```
Comment (ferrantim) [inline website/docs/feature-flag-tutorials/use-cases/user-management-access-controls.md:119]: I would like to add a link to Mateusz's blog on this topic. https://www.getunleash.io/blog/feature-flag-change-requests-how-to

Something like "For more detail on what enterprise-level change requests should include [read our blog](https://www.getunleash.io/blog/feature-flag-change-requests-how-to)

File: website/docs/feature-flag-tutorials/use-cases/user-management-access-controls.md [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+        "new_state": "disabled",
+        "environment": "production"
+    },
+    "source_ip": "192.168.1.100"
+}
+```
+
+The most advanced implementations integrate Unleash audit logs directly into broader Security Information and Event Management (SIEM) systems. This enables real-time monitoring and creates a holistic view of system interactions across multiple platforms. By correlating Unleash audit data with other enterprise security logs, organizations can detect subtle patterns and potential security anomalies that might go unnoticed in isolated systems.
+
+By implementing intelligent, automated, and granular access controls, organizations can create a secure foundation for continuous delivery at scale.
```
Comment (ferrantim) [inline website/docs/feature-flag-tutorials/use-cases/user-management-access-controls.md:172]: This conclusion sentence seems out of place.  Only refers to access controls, not the rest of the topics in the blog.

**Review by ferrantim — APPROVED** [review]
Looks good.  Still needs an engineering review but it is good from my perspective with the requested changes. Thanks!

File: website/docs/feature-flag-tutorials/use-cases/user-management-access-controls.md (hunk 3 above)
Comment (nnennandukwe) [reply website/docs/feature-flag-tutorials/use-cases/user-management-access-controls.md:106]: Definitions online say "at least 2 people need to approve" and yes, it is vague as to whether or not the requestor can be considered one of the "pairs of eyes".

However, looking a second time (post-holiday freshness :) ), I see that the [Open Risk Manual](https://www.openriskmanual.org/wiki/Four_Eyes_Principle) implies that 1 person needs to review, making it a total of 2 people involved.

File: website/docs/feature-flag-tutorials/use-cases/user-management-access-controls.md [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+1. Admin
+2. Editor
+3. Viewer
+
+_Project roles_:
+
+1. Owner
+2. Member
+
+Assign users with root roles to configure entire systems. These users can perform any operation within the Unleash platform and change permissions for other users when they belong to a specific user group.
```
[the remark this answers is not in the document]
Comment (nnennandukwe) [reply website/docs/feature-flag-tutorials/use-cases/user-management-access-controls.md:92]: The ability to perform all operations is just for admin roles.

Editors can do everything **except** managing users and roles. Editors can operate as project owners.

Viewers can operate as project members when added to a project. They can create, view, and update feature flags within a specific project.

**Review by dsusa72 — APPROVED** [review]
I focused my review in all sections that mention "audit". It looks good to me.

File: website/static/img/use-case-user-mgmt-cr.gif
Comment (melindafekete) [inline website/static/img/use-case-user-mgmt-cr.gif:1]: @nnennandukwe could you please remove any personal names when taking screenshots

**Review by melindafekete — DISMISSED** [review]
Just a few minor comments. I'll review once more when there is a build preview. Thanks!

**Review by gastonfournier — COMMENTED** [review]
Overall, looks good, just added a few comments
